### PR TITLE
Allow mocha-multi to be configured with an object via a mocharc file.

### PR DESCRIPTION
**Pull request by ghost**
Our configuration of the reporter requires passing an object, but
doing so via a mocharc file does not work because the option parser
only supports key=value pairs and any supplied object is trampled
before the reporter can is instantiated.

Solve this by allowing mocha-multi options to be specified using a
separate "mochaMultiOptions" key that can contain an object encoded
as JSON - this will be left alone by the mocha options parser and
thus the reporter is able to get hold of the necessary options.

### CODE REVIEW

File: mocha-multi.js [hunk 1]
```diff
@@ -293,6 +293,14 @@ function mochaMulti(runner, options) {
 
 class MochaMulti {
   constructor(runner, options) {
+    debugger;
```
Comment (jbdietrich) [inline mocha-multi.js]: I guess this is left over from development?

Comment (ghost) [reply mocha-multi.js]: 🤦

File: mocha-multi.js [hunk 2]
```diff
@@ -293,6 +293,14 @@ function mochaMulti(runner, options) {
...
+    if (options.mochaMultiOptions) {
+      let reporterOptions = options.mochaMultiOptions;
+      if (typeof reporterOptions === 'string') {
+        reporterOptions = JSON.parse(reporterOptions);
+      }
+      Object.assign(options, { reporterOptions });
```
Comment (papandreou) [inline mocha-multi.js]: Maybe output a warning of `options.reporterOptions` is already a non-empty object, as it'll be overwritten?

**Review by papandreou — APPROVED** [review]
LGTM 🌮

File: mocha-multi.js (hunk 2 above)
Comment (ghost) [reply mocha-multi.js]: Hmmmmm - my gut says if you've gone out of your way to use the "mochaMultiOptions" key then overwriting is the behaviour you're opting into so I din't think a warning makes much sense?

What I was just thinking though it a plain assignment might be better that `Object.assign()` though.

Comment (papandreou) [reply mocha-multi.js]: You're right that it's a perfectly reasonable outcome that `mochaMultiOptions` takes precedence, it just seems like a minor footgun.

Comment (ghost) [reply mocha-multi.js]: Think this is better as of: https://github.com/peakon/mocha-multi/pull/1/files#diff-a12c9c4ac2eeaf73fc7d38740c2da43a66cad7a74bcf4b95ef44955868f5a237R302

**Pull request merged**